src/main/java/com/example/waytogo/route/service/impl/RouteServiceJPA.java

```diff
@@ -0,0 +1,37 @@
+package com.example.waytogo.route.service.impl;
+
+import com.example.waytogo.route.model.entity.Route;
+import com.example.waytogo.route.repository.RouteRepository;
+import com.example.waytogo.route.service.api.RouteService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class RouteServiceJPA implements RouteService {
+    private final RouteRepository routeRepository;
+
+    @Override
+    public Optional<Route> getRouteById(UUID id) {
+        return routeRepository.findById(id);
+    }
+
+    @Override
+    public Route save(Route route) {
+        return routeRepository.save(route);
```

Comment:
ogolnie to tam widzialem ze w kontrollerze ustawiales id, ale jak mamy posta to losowe id sie daje i mi sie wydaje ze jak nie podasz to i tak zadziala, musisz w tym encji dac generatedId czy jakos tak wiesz ten generator id automatyczny to nie bedziesz musial ustawiac id